catalogue/webapp/components/WorkItemsStatus/WorkItemsStatus.js

```diff
@@ -0,0 +1,156 @@
+// @flow
+import Router from 'next/router';
+import fetch from 'isomorphic-unfetch';
+import { useEffect, useState, Fragment } from 'react';
+import { type Work } from '@weco/common/model/work';
+import Space from '@weco/common/views/components/styled/Space';
+import { Tag } from '@weco/common/views/components/Tags/Tags';
+import { classNames, font } from '@weco/common/utils/classnames';
+import Auth from '../Auth/Auth';
+
+type Props = {| work: Work |};
+
+type StacksLocation = {| id: string, label: string |};
+type StacksItemStatus = {| id: string, label: string |};
+type StacksItem = {|
+  id: string,
+  location: StacksLocation,
+  status: StacksItemStatus,
+|};
+type PhysicalLocations = {|
+  id: string,
+  items: StacksItem[],
+|};
+
+type ItemRequestButtonProps = {| item: StacksItem, workId: string |};
+const ItemRequestButton = ({ item, workId }: ItemRequestButtonProps) => {
+  function setRedirectCookie(workId: string, itemId: string) {
+    const searchParams = new URLSearchParams(window.location.search);
+    searchParams.set('action', `requestItem:/works/${workId}/items/${item.id}`);
+
+    const url = `${window.location.pathname}?${searchParams.toString()}`;
+    document.cookie = `WC_auth_redirect=${url}; path=/`;
+  }
+
+  async function requestItem(token: string) {
+    const request = await fetch(
+      `https://api.wellcomecollection.org/stacks/v1/requests`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: token,
+        },
+        body: JSON.stringify({ itemId: item.id }),
+      }
+    ).then(r => {
+      if (r.status === 200) return r.json();
+      console.error('invalid /requests');
+    });
+
+    return request;
+  }
+
+  useEffect(() => {
+    const action = Router.query.action;
+    if (action.startsWith('requestItem:')) {
+      const match = action.match(/requestItem:\/works\/[a-z0-9]+\/items\/(.*)/);
```

Comment:
Is it necessary to have both the `if` and the regex? Not 100% sure but think you could just put a `^` anchor at the start of the regex and it'd do the work of both statements. Also might be worth naming the regex to clarify what it does

Reply:
> Is it necessary to have both the if and the regex

Nope, I want to create a pattern for this where we can do it with other login actions so I'll pull it out as it's own thing when we actually implement the action TODO added